src/consensus/aft/raft_types.h

```diff
@@ -93,6 +95,25 @@ namespace aft
       }
       return nullptr;
     }
+
+    bool flag_enabled(kv::AbstractStore::Flag f) override
+    {
+      auto p = x.lock();
+      if (p)
+      {
+        return p->flag_enabled(f);
+      }
+      return false;
+    }
+
+    void unset_flag(kv::AbstractStore::Flag f) override
+    {
+      auto p = x.lock();
+      if (p)
+      {
+        return p->unset_flag(f);
+      }
+    }
```

Comment:
The `Adaptor` derives from `Store`, so it needs an implementation of the pure virtuals. I think it's better to define them instead of providing a dummy/empty implementation in the base class. I have no idea what kind of concept the `Adaptor` is supposed to adapt though, so I really can't tell whether these functions are out of place here.

Reply:
`aft::Store` (not `kv::Store`) is defined a few lines above the `Adaptor`. It is the limited interface from Raft to the `kv::Store`. Since Raft doesn't make used of these functions, I believe you don't need to define them in `aft::Store` at all?